pkg/gadgets/snapshot/process/types/process-collector.go

```diff
@@ -32,3 +35,91 @@ type Event struct {
 func GetColumns() *columns.Columns[Event] {
 	return columns.MustCreateColumns[Event]()
 }
+
+type nodeProcess struct {
+	event    *Event
+	children []*nodeProcess
+}
+
+func createTree(processes []*Event) (*nodeProcess, error) {
+	var root *nodeProcess
+
+	nodes := make(map[int]*nodeProcess, len(processes))
+	// Create a node for each processes.
+	for _, process := range processes {
+		nodes[process.Pid] = &nodeProcess{
+			event:    process,
+			children: make([]*nodeProcess, 0),
+		}
+	}
+
+	// Link all nodes together.
+	for _, node := range nodes {
+		ppid := node.event.ParentPid
+		if _, ok := nodes[ppid]; !ok {
+			if root != nil {
+				return nil, fmt.Errorf("container has two root process: %v and %v", root, node)
+			}
+
+			root = node
+
+			continue
+		}
+
+		nodes[ppid].children = append(nodes[ppid].children, node)
+	}
+
+	if root == nil {
+		return nil, fmt.Errorf("container has no root process")
+	}
+
+	return root, nil
+}
+
+func treeToString(node *nodeProcess) string {
+	var builder strings.Builder
+	treeToStringBuilder(node, &builder, 0)
+
+	return builder.String()
+}
+
+func treeToStringBuilder(node *nodeProcess, builder *strings.Builder, depth int) {
+	fmt.Fprintf(builder, "%s|-%s(%d)\n", strings.Repeat("\t", depth), node.event.Command, node.event.Pid)
+	if len(node.children) == 0 {
+		return
+	}
+
+	for _, child := range node.children {
+		treeToStringBuilder(child, builder, depth+1)
+	}
+}
+
+func PrintTree(processes []*Event) error {
```

Comment:
How about having an `io.Writer` parameter here and `fmt.Fprintf` to it instead of writing to stdout directly. Instead, `os.Stdout` can be given as argument in the `cmd/` packages calling PrintTree. That way third parties (and ourselves) can output the tree by other means.